### PR TITLE
Removing Argo

### LINKED ISSUE #90: Remove Argo?
Hi there!

I'm currently adding Heimdallr to our project and we already use [another mapping framework](https://github.com/lyft/mapper). Since dynamic frameworks add a considerable overhead when loading, I was think if it was possible to remove Argo as dependency here.

My main motivation is that the models seem to be simple enough to parse them manually instead and it would keep the library simpler, avoiding:

1. The overhead of another dynamic framework
2. Caring about if Argo updated to Swift X.Y
3. Potential conflicts between the Argo version used by this project and the user's project

This shouldn't even be a breaking change, since it's just an implementation detail.

I don't know the history, but it seems that on #16, everyone agreed that dependencies were fine, but https://github.com/trivago/Heimdallr.swift/commit/2be58cbe3d78903c282532ff0db8809a51db35e6 removed KeychainAccess in favor of a custom Keychain wrapper.

If response is positive to this issue, I'm willing to work on this and open a PR.

Thanks!

### PR DESCRIPTION
**Pull request by marcelofabri**
This fixes #90 

I wasn't able to figure out a way to make Carthage just remove Argo, so I had to run `carthage update ...`, which updated Nimble, Result and ReactiveCocoa. I hope it's not a problem.

### CODE REVIEW

**Comment by marcelofabri** [comment]
Has anybody had a chance to take a look at this? 😁

File: Heimdallr/Core/Heimdallr.swift [hunk 1]
```diff
@@ -155,10 +155,9 @@ public let HeimdallrErrorNotAuthorized = 2
                     completion(.Failure(error))
                 }
             } else {
-                switch OAuthError.decode(data!) {
-                case let .Success(error):
+                if let error = OAuthError.decode(data: data!) {
```
Comment (felixjendrusch) [inline Heimdallr/Core/Heimdallr.swift]: Would you mind manually unwrapping `data`, potentially failing by not satisfying a condition or using `fatalError` with an appropriate error message if `data` is `nil`?

Comment (marcelofabri) [inline Heimdallr/Core/Heimdallr.swift]: @felixjendrusch Done in 4ab6083

**Comment by hffmnn** [comment]
I would give it a go. I do not think that the dependency update should do any harm.

@felixjendrusch Green from your side?

And once more: Thank you @marcelofabri and a big 👍  for contributing.

**Comment by felixjendrusch** [comment]
@hffmnn yo. Thanks @marcelofabri.

**Pull request merged**